adafruit_rfm69.py

```diff
@@ -714,118 +740,170 @@ def send(
+            payload[3] = tx_header[3]
+        payload = payload + data
+        # Write payload to transmit fifo
+        self._write_fifo_from(payload)
         # Turn on transmit mode to send out the packet.
         self.transmit()
         # Wait for packet sent interrupt with explicit polling (not ideal but
         # best that can be done right now without interrupts).
         start = time.monotonic()
         timed_out = False
         while not timed_out and not self.packet_sent:
-            if (time.monotonic() - start) >= timeout:
+            if (time.monotonic() - start) >= self.xmit_timeout:
                 timed_out = True
-        # Listen again if necessary and return the result packet.
+        # Listen again if requested.
         if keep_listening:
             self.listen()
-        else:
-            # Enter idle mode to stop receiving other packets.
+        else:  # Enter idle mode to stop receiving other packets.
             self.idle()
-
         return not timed_out
 
+    def send_with_ack(self, data):
+        """Reliabe Datagram mode:
+           Send a packet with data and wait for an ACK response.
+           The packet header is automatically generated.
+           If enabled, the packet tranmsiion will be retried on failure
+        """
+        if self.ack_retries:
+            retries_remaining = self.ack_retries
+        else:
+            retries_remaining = 1
+        got_ack = False
+        self.sequence_number = (self.sequence_number + 1) & 0xFF
+        while not got_ack and retries_remaining:
+            self.identifier = self.sequence_number
+            self.send(data, keep_listening=True)
+            # Don't look for ACK from Broadcast message
+            if self.destination == _RH_BROADCAST_ADDRESS:
+                got_ack = True
+            else:
+                # wait for a packet from our destination
+                ack_packet = self.receive(timeout=self.ack_wait, with_header=True)
+                if ack_packet is not None:
+                    if ack_packet[3] & _RH_FLAGS_ACK:
+                        # check the ID
+                        if ack_packet[2] == self.identifier:
+                            got_ack = True
+                            break
+            # pause before next retry -- random delay
+            if not got_ack:
+                # delay by random amount before next try
+                time.sleep(self.ack_wait + self.ack_wait * random.random())
+            retries_remaining = retries_remaining - 1
+            # set retry flag in packet header
+            self.flags |= _RH_FLAGS_RETRY
+        self.flags = 0  # clear flags
+        return got_ack
+
+    # pylint: disable=too-many-branches
     def receive(
-        self,
-        timeout=0.5,
-        keep_listening=True,
-        with_header=False,
-        rx_filter=_RH_BROADCAST_ADDRESS,
+        self, keep_listening=True, with_ack=False, timeout=None, with_header=False
```

Comment:
```suggestion
        self, *, keep_listening=True, with_ack=False, timeout=None, with_header=False
```

Add a `*` so they are forced to be keyword arguments.

Reply:
Will do.

Reply:
Done